Source/Core/Common/OnionConfig.cpp

```diff
@@ -0,0 +1,380 @@
+						return true;
+			} while (--layers_it != m_layers.end());
+
+			return false;
+		}
+
+		bool RecursiveOnionPetal::Get(const std::string& key, std::string* value, const std::string& default_value) const
+		{
+			auto layers_it = m_layers.find(OnionLayerType::LAYER_META);
+
+			if (layers_it == m_layers.begin())
+				return OnionPetal::Get(key, value, default_value);
+
+			while (--layers_it != m_layers.begin());
+			{
+				OnionPetal* layer_petal = layers_it->second->GetPetal(m_system, m_name);
+				if (layer_petal)
+					if (layer_petal->Exists(key))
+						return layer_petal->Get(key, value, default_value);
+			}
+
+			return OnionPetal::Get(key, value, default_value);
+		}
+
+		void RecursiveOnionPetal::Set(const std::string& key, const std::string& value)
+		{
+			_assert_msg_(COMMON, false, "Don't try to set values here!");
+		}
+
+		class RecursiveBloomLayer final : public BloomLayer
+		{
+		public:
+			RecursiveBloomLayer() : BloomLayer(OnionLayerType::LAYER_META) {}
+			OnionPetal* GetOrCreatePetal(OnionSystem system, const std::string& petal_name) override;
+		};
+
+		OnionPetal* RecursiveBloomLayer::GetOrCreatePetal(OnionSystem system, const std::string& petal_name)
+		{
+			OnionPetal* petal = GetPetal(system, petal_name);
+			if (!petal)
+			{
+				m_petals[system].emplace_back(new RecursiveOnionPetal(m_layer, system, petal_name));
+				petal = m_petals[system].back();
+			}
+			return petal;
+		}
+	}
+
+		bool OnionPetal::Exists(const std::string& key) const
+		{
+			const auto& it = m_values.find(key);
+			if (it != m_values.end())
+				return true;
+			else
+				return false;
+		}
+
+		bool OnionPetal::Delete(const std::string& key)
+		{
+			auto it = m_values.find(key);
+			if (it == m_values.end())
+				return false;
+
+			m_values.erase(it);
+			return true;
+		}
+
+		void OnionPetal::Set(const std::string& key, const std::string& value)
+		{
+			auto it = m_values.find(key);
+			if (it != m_values.end())
+				it->second = value;
+			else
+				m_values[key] = value;
+		}
+
+		void OnionPetal::Set(const std::string& key, u32 newValue)
+		{
+			OnionPetal::Set(key, StringFromFormat("0x%08x", newValue));
+		}
+
+		void OnionPetal::Set(const std::string& key, float newValue)
+		{
+			OnionPetal::Set(key, StringFromFormat("%#.9g", newValue));
+		}
+
+		void OnionPetal::Set(const std::string& key, double newValue)
+		{
+			OnionPetal::Set(key, StringFromFormat("%#.17g", newValue));
+		}
+
+		void OnionPetal::Set(const std::string& key, int newValue)
+		{
+			OnionPetal::Set(key, StringFromInt(newValue));
+		}
+
+		void OnionPetal::Set(const std::string& key, bool newValue)
+		{
+			OnionPetal::Set(key, StringFromBool(newValue));
+		}
+
+		void OnionPetal::Set(const std::string& key, const std::vector<std::string>& newValues)
+		{
+			std::string temp;
+			// Join the strings with ,
+			for (const std::string& value : newValues)
+			{
+				temp = value + ",";
+			}
+			// remove last ,
+			temp.resize(temp.length() - 1);
```

Comment:
I wonder if there is a more elegant solution to imploding a vector into a string (like C#'s `string.Join(",", newValues)`).

But either way, since you're only replacing `temp`, nothings going to happen there.
I think you intended `temp += value + ",";` (with a `+=`), right?